### ISSUE TITLE
GANs L1.5 Tips and Tricks

### ISSUE THREAD

**Issue by sahiljuneja** [udacity/dlnd-issue-reports#271]
This is a long post. But before I begin, this was a really good lesson as per me. I do point out how some more details could be added, but from a perspective of a broad overview, this was quite good! Thank you!

This lesson discusses a few terms/topics which I don't think have been introduced before. [Correct me if I'm mistaken]

- Universal Approximator.
- Leaky ReLUs
- Label smoothing strategy for regularization [Perhaps a link in one of the prior lessons as an extra resource instead of a separate section for your convenience]
- Small clarification on what an "op" is. Although shouldn't be needed if the students have gone through any parts of TF documentation.

Some discussions which can benefit students I believe, unless you prefer the discussions to be over the forums or Slack (which si fine).

- Why tanh is a popular choice for the output of the generator?
- Why sigmoid and not softmax for discriminator output?
- Why is Adam Optimizer preferred? [A resource for that might be good enough]
- Why the "multiplying labels with a value slightly smaller than 1" strategy helps the discriminator to generalize better?
- When discussing the shape of the input to the GANs with convolutions, perhaps a small animation in the video to showcase the inputs just as a reminder for the students?
- The last part of the video on batch normalization. Does seem to miss out on some intuitive explanations on "why" we do this or why it's preferred as Ian puts it.
- A link to the DC-GAN paper since it's referenced so often, or related tutorials.Also for the "Improved Techniques for training GANs" paper Ian mentions.

Although, I completely understand why you wouldn't want to go in-depth with the above given it's a Foundations program. But thought of bringing it up since Ian does go into some brief explanations on some other subtopics in this video.

Some other issues related to animations which I think will be on your list of improvements already in the long run -

- 5:29 to 5:45 the animation doesn't seem to properly support the dialogue. I think due to lack of time you guys went with reusing some initial videos. Could these be improved later to match with the dialogue? Or you think it's good enough?
- The explanation animation of increasing the size of feature maps by using convolution transpose ops with stride greater than 1. Couldn't visually understand clearly what was happening.
